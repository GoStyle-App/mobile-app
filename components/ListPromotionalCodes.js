import React, { useState } from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import getPromotionalCodes from '../api/getPromotionalCodes';
import PromotionalCode from './PromotionalCode';
import { Colors, IconButton } from 'react-native-paper';

export default function ListPromotionalCodes({ navigation }) {
    const [ codes, setCodes ] = useState([]);

    getPromotionalCodes().then((promotionalCodes) => {
        let codes = [];

        promotionalCodes.forEach(promotionalCode => {
            codes.push({
                'id': promotionalCode.id,
                'code': promotionalCode.code,
                'label': promotionalCode.label,
                'description': promotionalCode.description,
                'startDate': promotionalCode.startDate,
                'endDate': promotionalCode.endDate,
            });
        });

        setCodes(codes);
    });

    if (codes.length > 0) {
        return (
            <View style={ styles.codes }>
                <Text style={ styles.title }>Liste des codes promotionnels scanables</Text>
                <FlatList
                    data={ codes }
                    keyExtractor={(item) => item.id.toString()}
                    renderItem={({ item }) =>
                        <PromotionalCode
                            code={ item }
                            navigation={ navigation }
                            myCodes={ codes }
                            detail={ false }
                        />
                    }
                />
                <View style={ styles.buttonContainer }>
                    <IconButton
                        icon="qrcode-scan"
                        color={ Colors.white }
                        size={ 35 }
                        onPress={() => {
                            navigation.navigate('Scan');
                        }}
                    />
                </View>
            </View>
        );
    }

    return (
        <View style={ styles.codes }>
            <Text style={ styles.title }>Liste des codes promotionnels scanables</Text>
            <Text style={ styles.text }>Pas de codes promos</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    codes: {
        marginTop: 80,
        alignItems: 'center',
    },
    title: {
        marginBottom: 30,
        color: '#2E2E2E',
        fontSize: 20,
        textTransform: 'uppercase',
        textAlign: 'center',
    },
    buttonContainerSimple: {
        width: 150,
    },
    buttonContainer: {
        backgroundColor: '#38B6FF',
        borderRadius: 50,
        marginLeft: 250,
        marginTop: 350,
    },
});
