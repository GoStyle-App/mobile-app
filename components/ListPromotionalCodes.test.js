import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import 'isomorphic-fetch';
import ListPromotionalCodes from './ListPromotionalCodes';


/* global describe, expect, test */
describe('Testing List Promotional Codes component', () => {
    const route = {
        'key': 'Home-Nx8Rj_yGS8oGfmw-QfVCq',
        'name': 'Home',
        'params': {
            'link': ''
        },
    };
    const navigation = {
        'addListener': function addListener(){},
        'canGoBack': function canGoBack(){},
        'closeDrawer': function anonymous(){},
        'dangerouslyGetParent': function dangerouslyGetParent(){},
        'dangerouslyGetState': function anonymous(){},
        'dispatch': function dispatch(){},
        'goBack': function anonymous(){},
        'isFocused': function isFocused(){},
        'jumpTo': function anonymous(){},
        'navigate': function anonymous(){},
        'openDrawer': function anonymous(){},
        'removeListener': function removeListener(){},
        'reset': function anonymous(){},
        'setOptions': function setOptions(){},
        'setParams': function anonymous(){},
        'toggleDrawer': function anonymous(){},
    };

    test('The app prints the corresponding screen', async () => {
        const component = (
            <NavigationContainer>
                <ListPromotionalCodes navigation={ navigation } route={ route } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const screen = await findByText('Liste des codes promotionnels scanables');

        expect(screen).toBeTruthy();
    });

    test('Correct render of component', () => {
        const component = render(<ListPromotionalCodes navigation={ navigation } route={ route } />);

        expect(component).toMatchSnapshot();
    });
});
