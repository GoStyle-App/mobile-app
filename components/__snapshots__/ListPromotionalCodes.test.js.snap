// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List Promotional Codes component Correct render of component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2E2E2E",
        "fontSize": 20,
        "marginBottom": 30,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Liste des codes promotionnels scanables
  </Text>
  <Text>
    Pas de codes promos
  </Text>
</View>
`;
